File: sdk/python/examples/openai/01_basic_agent.py
# Copyright (c) 2025 AgentSpan
# Licensed under the MIT License. See LICENSE file in the project root for details.

"""Basic OpenAI Agent — simplest possible agent with no tools.

Demonstrates:
    - Defining an agent using the OpenAI Agents SDK
    - Running it on the Conductor agent runtime (auto-detected)
    - The runtime serializes the agent generically and the server
      normalizes the OpenAI-specific config into a Conductor workflow.

Requirements:
    - pip install openai-agents
    - Conductor server with OpenAI LLM integration configured
    - export AGENTSPAN_SERVER_URL=http://localhost:7001/api
"""

from agents import Agent

from agentspan.agents import AgentRuntime

agent = Agent(
    name="greeter",
    instructions="You are a friendly assistant. Keep your responses concise and helpful.",
    model="gpt-4o",
)

with AgentRuntime() as runtime:
    result = runtime.run(agent, "Say hello and tell me a fun fact about the Python programming language.")
    result.print_result()

File: sdk/typescript/examples/openai/01-basic-agent.ts
/**
 * Basic OpenAI Agent -- simplest possible agent with no tools.
 *
 * Demonstrates:
 *   - Defining an agent using the OpenAI Agents SDK mock
 *   - Running it on the Conductor agent runtime (auto-detected)
 *   - The runtime serializes the agent generically and the server
 *     normalizes the OpenAI-specific config into a Conductor workflow.
 *
 * Requirements:
 *   - Conductor server with OpenAI LLM integration configured
 *   - AGENTSPAN_SERVER_URL=http://localhost:8080/api as environment variable
 *   - AGENTSPAN_LLM_MODEL=openai/gpt-4o-mini as environment variable
 */

import { AgentRuntime } from '../../src/index.js';
import { llmModel } from '../settings.js';

// Mock OpenAI Agent -- duck-typing detection: has .run() + .tools + .model
const agent = {
  run: async (prompt: string) => ({ output: `Greeted: ${prompt}` }),
  tools: [],
  model: llmModel,
  name: 'greeter',
  instructions: 'You are a friendly assistant. Keep your responses concise and helpful.',
  _openai_agent: true,
};

const runtime = new AgentRuntime();
try {
  const result = await runtime.run(
    agent,
    'Say hello and tell me a fun fact about the Python programming language.',
  );
  result.printResult();
} finally {
  await runtime.shutdown();
}
